src/lib/storage/table.cpp

```diff
@@ -267,6 +268,40 @@ void Table::set_table_statistics(const std::shared_ptr<TableStatistics>& table_s
 
 std::vector<IndexStatistics> Table::indexes_statistics() const { return _indexes; }
 
+const std::vector<TableConstraintDefinition>& Table::get_unique_constraints() const { return _constraint_definitions; }
+
+void Table::add_unique_constraint(const std::vector<ColumnID>& column_ids, bool primary) {
+  for (const auto& column_id : column_ids) {
+    Assert(column_id < column_count(), "ColumnID out of range");
+    Assert(!primary || !column_is_nullable(column_id), "Column must be not nullable for primary key constraint");
+  }
+
+  Assert(column_ids.size() == std::set<ColumnID>(column_ids.begin(), column_ids.end()).size(),
+         "Column IDs must be unique");
+
+  {
```

Comment:
Does that scope really change anything? It's not running multiple times and the scope will not be left any earlier?

Reply:
No, but good practice requires an explicit scope for scoped locks. If someone were to add something to the method, chances are that they would inadvertently prolong the lifetime of the scoped_lock.